src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -2868,6 +2868,25 @@ private class ConsoleHostStartupException : Exception
         private System.Threading.Thread _breakHandlerThread;
         private bool _isDisposed;
         internal ConsoleHostUserInterface ui;
+
+#if CORECLR
+        internal Lazy<TextReader> ConsoleIn { get; } = new Lazy<TextReader>(() => Console.In);
+#else
+        internal Lazy<TextReader> ConsoleIn { get; } = new Lazy<TextReader>(() =>
+        {
+            // This is a workaround for a full clr issue that causes a hang when calling PowerShell from ruby.
+            // They use named pipes instead of anonymous pipes, and for some reason that triggers a hang
+            // reading from Console.In.
+            var inputHandle = ConsoleControl.NativeMethods.GetStdHandle(-10);
```

Comment:
Is it not obvious that it's standard in? (std handle, assigned to var named inputHandle, and comment mentions Console.In)